### PR TITLE
A singleton for logging

### PR DESCRIPTION
**Pull request by micheldebree**
A simple abstraction for logging diagnostic messages.
Implementation uses SDL_Log now.

### CODE REVIEW

File: SIDFactoryII/source/foundation/sound/audiostream.cpp [hunk 1]
```diff
@@ -49,7 +49,26 @@ namespace Foundation
 
 		SDL_AudioSpec audio_spec_created;
 
+		const int count = SDL_GetNumAudioDevices(0);
+
+		for (int i = 0; i < count; ++i)
+		{
+			Utility::Logging::instance().Info("Audio device %d: %s", i, SDL_GetAudioDeviceName(i, 0));
```
Comment (micheldebree) [inline SIDFactoryII/source/foundation/sound/audiostream.cpp:56]: This is not logging the right integer value..

Comment (micheldebree) [reply SIDFactoryII/source/foundation/sound/audiostream.cpp:56]: Fixed.

File: SIDFactoryII/main.cpp [hunk 2]
```diff
@@ -25,14 +26,29 @@ void BuildResource();
...
+
+	Utility::Logging::instance().Info("SIDFactoryII %s build %s", os, build_number);
+
 	// Initialize SDL
 	const int sdl_init_result = SDL_Init(SDL_INIT_TIMER | SDL_INIT_AUDIO | SDL_INIT_VIDEO);
 	if (sdl_init_result < 0)
 	{
-		std::cout << "SDL initialization failed. SDL Error: " << SDL_GetError();
-
+		std::cout << "SDL initialization failed. SDL Error:" << SDL_GetError();
```
Comment (TheRealLaxity) [inline SIDFactoryII/main.cpp:51]: The extra space was there for formatting the ouput so that there's a space after the colon.

**Review by TheRealLaxity — APPROVED** [review]
I'm good with this, but it's imperative that the logger is only used from the main thread, or we'd have to explicitely construct the singleton instance with an internal mutex.

File: SIDFactoryII/main.cpp (hunk 2 above)
Comment (micheldebree) [reply SIDFactoryII/main.cpp:51]: My mistake, I meant to put back the original code after I changed this to use the logging singleton. Didn't think like a good idea to use SDL logging if SDL couldn't be initialized.

Comment (micheldebree) [reply SIDFactoryII/main.cpp]: Fixed.

**Comment by micheldebree** [comment]
> I'm good with this, but it's imperative that the logger is only used from the main thread, or we'd have to explicitely construct the singleton instance with an internal mutex.

I added a little N.B. to the header file to only use it from main thread.

**Pull request merged**